blocks/bricks/cost.py

```diff
@@ -58,7 +58,8 @@ def cost_matrix(self, y, y_hat):
 class CategoricalCrossEntropy(Cost):
     @application(outputs=["cost"])
     def apply(self, y, y_hat):
-        cost = tensor.nnet.categorical_crossentropy(y_hat, y).mean()
+        cost = tensor.nnet.categorical_crossentropy(y_hat, y).mean(
+            acc_dtype=floatX)
```

Comment:
It resolved, for me, the same error messages I got in MisclassificationRate regarding float64 and CUDA. I guess this happens when y is int64.